### ISSUE TITLE
Style/SafeNavigation adding broken comments, no output to console regarding this.

### ISSUE THREAD

**Issue by fedeaux** [rubocop-hq/rubocop#7397]
## Expected behavior

I run `rubocop -a some/file.rb`, I expect some violations to be automatically corrected.

## Actual behavior

The violations are automatically corrected but it takes a multiline comment and replicates between the "ifs" and their conditions

## Steps to reproduce the problem
`some/file.rb` example:

Before

```
class SomeController < ApplicationController
  skip_before_action :verify_authenticity_token

  before_action :authenticate_user!, :authenticate_role, except: [
    :check,
    :check_codes,
    :fetch_actionable]

  # returns guest.credits based on total and dates
  # decorates with balance and pretty_name
  # includes credits total
  def fetch_actionable
    current_daily_pricing = params[:dailyRates].values

    if current_daily_pricing.all? { |r| r != 'x' }
      guest = Guest.find_by(email: params[:email])
      @credits = Credit::Calculator.new(guest.credits).fetch_actionable(
        params[:check_in],
        params[:check_out],
        params[:dailyRates],
        params[:email],
        params[:subtotal]
      )

      if guest && guest.valid_token?(params[:access_token], params[:client]) && @credits
        @credits_total = 0
        @credits_total = @credits.inject(0) do |sum, credit|
          next unless credit[:balance]
          sum + credit[:balance]
        end

        render json: {
          updated_credits: @credits,
          credits_total:   @credits_total
        }
      end

    else
      render json: {
        error: 'no_availability'
      }
    end
  end
end
```

after: (Notice the duplicated comment between the "if" and it's condition
```
class SomeController < ApplicationController
  skip_before_action :verify_authenticity_token

  before_action :authenticate_user!, :authenticate_role, except: %i[
    check
    check_codes
    fetch_actionable
  ]

  # returns guest.credits based on total and dates
  # decorates with balance and pretty_name
  # includes credits total
  def fetch_actionable
    current_daily_pricing = params[:dailyRates].values

    if current_daily_pricing.all? { |r| r != 'x' }
      guest = Guest.find_by(email: params[:email])
      @credits = Credit::Calculator.new(guest.credits).fetch_actionable(
        params[:check_in],
        params[:check_out],
        params[:dailyRates],
        params[:email],
        params[:subtotal]
      )

      if # returns guest.credits based on total and dates
# decorates with balance and pretty_name
# includes credits total
guest&.valid_token?(params[:access_token], params[:client]) && @credits
        @credits_total = 0
        @credits_total = @credits.inject(0) do |sum, credit|
          next unless credit[:balance]

          sum + credit[:balance]
        end

        render json: {
          updated_credits: @credits,
          credits_total: @credits_total
        }
      end

    else
      render json: {
        error: 'no_availability'
      }
    end
  end
end
```

## RuboCop version
0.75.0 (using Parser 2.6.5.0, running on ruby 2.6.5 x86_64-darwin16)
(Also tried with 0.69)

## .rubocop.yml

```
Metrics/LineLength:
  Enabled: false

Metrics/ClassLength:
  Enabled: false

Style/FrozenStringLiteralComment:
  Enabled: false

Style/ClassAndModuleChildren:
  Enabled: false
```

If I add

```
Style/SafeNavigation
  Enabled: false
```

The problem disappears

**Comment by buehmann** [comment]
```
74a70713bb87f40ff0fae2e98fefb751a8d41bb9 is the first bad commit
commit 74a70713bb87f40ff0fae2e98fefb751a8d41bb9
Author: Tejas Bubane <tejasbubane@gmail.com>
Date:   Thu Jun 27 00:14:59 2019 +0530

    [Fix #7161] Fix `Style/SafeNavigation` cop for preserve comments inside if expression
    
    Closes #7161.
```
This is caused by https://github.com/rubocop-hq/rubocop/pull/7179. @tejasbubane Could you take a look at this?

**Comment by buehmann** [comment]
Here is a smaller example:
```ruby
# comment
def foo
  guest && guest.valid_token?
end
```
is auto-corrected to
```ruby
# comment
def foo
  # comment
guest&.valid_token?
end
```
```
Offenses:

7397.rb:3:3: C: [Corrected] Style/SafeNavigation: Use safe navigation (&.) instead of checking if an object exists before calling the method.
  guest && guest.valid_token?
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
```
